pkg/cmd/adm/register_member.go

```diff
@@ -351,35 +480,55 @@ func (v *registerMemberValidated) confirmationPrompt() ioutils.ConfirmationMessa
 	return ioutils.WithMessagef(sb.String(), args...)
 }
 
-func (v *registerMemberValidated) perform(ctx *extendedCommandContext, newCommand client.CommandCreator) error {
+func (v *registerMemberValidated) perform(ctx *extendedCommandContext) error {
+	// wait for the toolchaincluster-member SA to be ready
+	toolchainClusterSAKey := runtimeclient.ObjectKey{
+		Name:      "toolchaincluster-member",
+		Namespace: v.args.memberNamespace,
+	}
+	if err := waitForToolchainClusterSA(ctx.CommandContext, v.memberClusterData.client, toolchainClusterSAKey, v.args.waitForReadyTimeout); err != nil {
+		ctx.Println("The toolchaincluster-member ServiceAccount in the member cluster is not present.")
+		ctx.Printlnf("Please check the %s ToolchainCluster ServiceAccount in the %s member cluster.", toolchainClusterSAKey, v.memberClusterData.apiEndpoint)
+		return err
+	}
 	// add the host entry to the member cluster first. We assume that there is just 1 toolchain cluster entry in the member
 	// cluster (i.e. it just points back to the host), so there's no need to determine the number of entries with the same
 	// API endpoint.
+	if err := v.addCluster(ctx, toolchainClusterSAKey, configuration.Host); err != nil {
```

Comment:
the reason why the e2e tests are failing is that you use member SA (in member namespace) when joining host cluster (and the other way around a few lines below when joining member cluster). As a result, host fails when it tries to get member resources because it uses host SA that doesn't have permission for member resources.
The namespace name is already part of the `clusterData` so you can construct the expected ObjectKey of the SA (and wait for the readiness) inside `addCluster` method.

Reply:
great catch ! thanks also for your suggestion, I've hopefully did it well this time, please see https://github.com/kubesaw/ksctl/pull/53/commits/586a65903a81dfa1584260f5aa00f71d1db744a6